spec/models/profile_spec.rb

```diff
@@ -40,4 +40,27 @@
       expect(profile.uuid).to be_present
     end
   end
+
+  describe "validations" do
+    it "validates github url" do
+      profile.github_url = "github.com"
+      expect(profile).not_to be_valid
+      profile.github_url = "https://github.com"
+      expect(profile).to be_valid
+    end
+
+    it "validates linkedin url" do
+      profile.linkedin_url = "linkedin.com"
+      expect(profile).not_to be_valid
+      profile.linkedin_url = "https://linkedin.com"
+      expect(profile).to be_valid
+    end
+
+    it "validates twitter url" do
+      profile.twitter_url = "x.com"
+      expect(profile).not_to be_valid
+      profile.twitter_url = "https://x.com"
+      expect(profile).to be_valid
+    end
```

Comment:
When sharing your profile from the respective mobile apps, the generated link always contains https so i agree with this approach. Otherwise we would need to generate our own regex validation as the gem works like this

Reply:
I think we should make the application as easy to use as possible. Therefore, we should add the scheme, https, if a user inputs a URL without one. After that, we should run the validations. In this case, the spec should be green because we would add the scheme via a callback if one wasn't given

Reply:
We are now setting https scheme before validation.